Documentation/ceph-client-crd.md

```diff
@@ -0,0 +1,47 @@
+---
+title: Client CRD
+weight: 3500
+indent: true
+---
+
+# Ceph Client CRD
+
+Rook allows creation and updating clients through the custom resource definitions (CRDs).
+For more information about user management and capabilities see the [Ceph docs](https://docs.ceph.com/docs/master/rados/operations/user-management/).
+
+## Use Case
+
+Use Client CRD in case you want to integrate Rook with with applications that are using LibRBD directly.
+For example for OpenStack deployment with Ceph backend use Client CRD to create OpenStack services users.
+
+Don't use Client CRD for Flex or CSI driver users, they are created automatically.
+
+## Creating Ceph User
+
+To get you started, here is a simple example of a CRD to configure a Ceph client with capabilities.
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephClient
+metadata:
+  name: glance
+  namespace: rook-ceph
+spec:
+  caps:
+    mon: 'profile rbd'
+    osd: 'profile rbd pool=images'
+---
+apiVersion: ceph.rook.io/v1
+kind: CephClient
+metadata:
+  name: cinder
+  namespace: rook-ceph
+spec:
+  caps:
+    mon: 'profile rbd'
+    osd: 'profile rbd pool=volumes, profile rbd pool=vms, profile rbd-read-only pool=images'
```

Comment:
Sage had some comments about this here: https://github.com/rook/rook/issues/3260#issuecomment-498730717

Do we want to "leak" Ceph's permissions structure through Rook?

Reply:
@mateuszlos Should we consider making this more generic as asked by Blaine and Sage? I also noticed below that currently this would only support caps for mon, mgr, osd, and mds. There are other types of caps as well. If we made this more generic and just passed it on to ceph, rook wouldn't have to worry about which caps to support. What do you think? 
(apologies on the delay, it would be great to get this finished up)

Reply:
Ok I will provide updates in next few days

Reply:
@mateuszlos How is it looking? Note that we have feature freeze for v1.2 coming up in the next week.

Reply:
@travisn updated, please take a look